sys/phydat/phydat.c

```diff
@@ -61,7 +61,7 @@ static const uint32_t divisors[] = {
 
 void phydat_fit(phydat_t *dat, const int32_t *values, unsigned int dim)
 {
-    assert(dim <= ARRAY_SIZE(dat->val));
+    assert(dim <= (sizeof(dat->val)/sizeof(dat->val[0])));
```

Comment:
Mhh curious that this isn't caught, but fixing this is not really part of this PR.